### DISCUSSION TITLE
I'm looking for a way to dynamically swap between Win/Linux and Mac layouts, i.e.

### DISCUSSION THREAD

**Discussion by manna-harbour** [manna-harbour/miryoku#192] [Q&A] [answered]
_Originally posted by @partap in https://github.com/manna-harbour/miryoku/discussions/85#discussioncomment-5086780_

        I'm looking for a way to dynamically swap between Win/Linux and Mac layouts, i.e.

- Swap clipboard keycodes between using Ctrl and Cmd (like MIRYOKU_CLIPBOARD=MAC, but dynamic)
- Swap Ctrl and Cmd modifier positions in general (in QMK, I use CG_SWAP/CG_NORM for that, but not sure how with ZMK)

Just starting out with zmk (and miryoku_zmk) on a new Corne keyboard...
On my "old" Iris, I've been using QMK and a custom keymap that's been moving closer and closer to a full reimplementation of miryaku.  My main issue with the Iris is the teeny tiny memory size (and no bluetooth)

**Comment by manna-harbour** [comment, marked as answer]
@partap
See https://github.com/manna-harbour/miryoku/issues/33.

You can substitute mods with a macOS setting, but that won't help with dynamic changing of clipboard keycodes.

ZMK doesn't have any support for changing modes between Win and Mac, but you can emulate it with extra layers. You can add and substitute layers using https://github.com/manna-harbour/miryoku/discussions/85. You'd add copies of all the half layers with the new mod order and modified clipboard bindings, and replace Extra with a copy of Base that has the new mod order and uses the new half layers on the thumb layer-taps, then [switch between Base and Extra](https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features) to toggle.

**Reply by partap** [reply]
ok, thanks...
I swapped ctrl/cmd with the macos keyboard settings for now.
I'm not sure why the clipboard keys wouldn't work, if the OS is just translating any ctrl input into cmd, but you're right, they don't.  I guess they are sending different codes than just ctrl-X/C/V...

I'll look into adding new layers when I get a chance...I still haven't gotten around to exploring zmk much yet.

**Reply by manna-harbour** [reply]
Try the `win` clipboard option. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#clipboard and https://github.com/manna-harbour/miryoku_zmk/blob/master/miryoku/miryoku_clipboard.h.
